### DISCUSSION TITLE
How can I dipose resurces in an effect class?

### DISCUSSION THREAD

**Discussion by Martin-Andersen** [mrpmorris/Fluxor#406] [Q&A]
I have a SignalR hub in a effect class that I want to dispose properly. But how? 

Created [this gist](https://gist.github.com/Martin-Andersen/6047a68025366644104e5ecd8a154563) so we have some concrete to talk about

**Comment by mrpmorris** [comment]
Why are you creating the hub instead of having it injected?

If you inject it, then the container that created it will dispose of it.

**Reply by Martin-Andersen** [reply]
You need to create and hold on to the connection on the client side

public class PriceUpdatedEffects: IAsyncDisposable, IDisposable
{
    private readonly HubConnection _hubConnection;

    public PriceUpdatedEffects(IConfiguration configuration)
    {
        var hubUrl = configuration["PriceUpdatedHub"];
        _hubConnection = new HubConnectionBuilder()
            .WithUrl($"{hubUrl}/priceUpdatedHub")
            .WithAutomaticReconnect()
            .Build();
    }

**Reply by mrpmorris** [reply]
You should be able to implement IDisposable or IDisposableAsync and the DI container will call that when it disposes of your effect.
